src/tools/clippy/clippy_lints/src/functions.rs

```diff
@@ -516,7 +516,7 @@ fn check_needless_must_use(
                 );
             },
         );
-    } else if !attr.is_value_str() && is_must_use_ty(cx, return_ty(cx, item_id)) {
+    } else if !attr.value_str().is_some() && is_must_use_ty(cx, return_ty(cx, item_id)) {
```

Comment:
Doesn't this imply that `is_value_str` *is* used, just not by `compiler/*`?

Reply:
See https://github.com/rust-lang/rust/pull/83185#discussion_r594888909 - it seems weird to maintain this in-tree when it's so easy to write `.is_some()`.